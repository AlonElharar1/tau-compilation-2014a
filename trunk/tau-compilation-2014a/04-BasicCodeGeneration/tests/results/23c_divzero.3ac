	goto :_ic_main
:_ic_main
	= 0 $2
	param :str1
	call :println
	if $2 :endCheck2
	param :str3
	call :println
	param 1
	call :exit
:endCheck2
	/ 5 $2 $3
	= $3 $2
	param $2
	call :printi
	param :str4
	call :println
	param :str5
	call :println
	ret
.data
:str1
	8
	"5 / b = "
:str3
	32
	"Runtime Error: Division by zero!"
:str4
	0
	""
:str5
	34
	"Should not print this or get here."
