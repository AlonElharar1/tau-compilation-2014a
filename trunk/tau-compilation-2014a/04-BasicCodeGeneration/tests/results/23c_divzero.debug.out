pc = 0    @   goto :_ic_main
pc = 1    @   = 0 $2
 $0 = 0
 $1 = 0
 $2 = 0
pc = 2    @   param :str1
 $0 = 0
 $1 = 0
 $2 = 0
pc = 3    @   call :println
5 / b = 
 $0 = 0
 $1 = 0
 $2 = 0
pc = 4    @   if $2 :endCheck2
 $0 = 0
 $1 = 0
 $2 = 0
pc = 5    @   param :str3
 $0 = 0
 $1 = 0
 $2 = 0
pc = 6    @   call :println
Runtime Error: Division by zero!
 $0 = 0
 $1 = 0
 $2 = 0
pc = 7    @   param 1
 $0 = 0
 $1 = 0
 $2 = 0
pc = 8    @   call :exit
 $0 = 0
 $1 = 0
 $2 = 0
HALTED
 $0 = 0
 $1 = 0
 $2 = 0
